bin/util.py

```diff
@@ -411,6 +411,15 @@ def resolve_path_argument(problem, path, type, suffixes=[]):
     return None
 
 
+def shorten_path(problem, path):
```

Comment:
ah this is neat! Can you explain in a comment?

Normally I'd favour a more structured path, but since that path already contains a hash replacing it by a single shorter hash sgtm